src/app/Dependencies.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useCallback, useEffect } from 'react';
+import { jsx } from '@emotion/core';
+import ReactFlow, { ReactFlowProvider, addEdge, removeElements, isNode } from 'react-flow-renderer';
+import * as dagre from 'dagre';
+import { v4 as uuid } from 'uuid';
+import { getDependencyMap } from './commands';
+import { useStore } from './store';
+
+export const Dependencies = () => {
+  const position = { x: 0, y: 0 };
+  const edgeType = 'smoothstep';
+  const test = useStore(m => m.state.dependencyMap);
+
+  const dependencies = {
+    pilet1: ['react-router', 'react', 'foo@1.2.3'],
+    pilet2: ['react', 'foo@1.2.3'],
+  };
+
+  console.log(test)
+  console.log(dependencies)
+
+  // const dependencies = {
+  //   pilet1: [{ desired: 'react-router', resolved: 'react-router' }, { desired: 'react', resolved: 'react' }, { desired: 'foo@1.2.3', resolved: 'foo@1.2.3' }],
+  //   pilet2: [{ desired: 'react', resolved: 'react' }, { desired: 'foo@1.2.3', resolved: 'foo@1.2.3' }],
+  // };
+
+  let nonDuplicatedDependencies = [];
+
+  const initialElements = [
+    {
+      id: 'Piral',
+      data: { label: 'Piral' },
+      position,
+    },
+    ...Object.keys(dependencies).map(
+      pilet => (
+        dependencies[pilet].map(dep => nonDuplicatedDependencies.push(dep.resolved || dep)),
+        {
+          id: pilet,
+          type: 'input',
+          data: { label: pilet },
+          position,
+        }
+      ),
+    ),
+    ...Object.keys(dependencies).map(pilet => ({
+      id: pilet,
+      source: 'Piral',
+      target: pilet,
+      type: edgeType,
+      animated: true,
+    })),
+    ...Array.from(new Set(nonDuplicatedDependencies)).map(depName => ({
+      id: depName,
+      type: 'output',
+      data: { label: depName },
+      position,
+    })),
+    ...Object.keys(dependencies).flatMap(pilet =>
+      dependencies[pilet].map(depName => ({
+        id: uuid(),
+        source: pilet,
+        target: depName.resolved || depName,
+        type: 'straight',
+        animated: false,
+        label: depName.resolved || depName,
+      })),
+    ),
+  ];
+
+  const dagreGraph = new dagre.graphlib.Graph();
+  dagreGraph.setDefaultEdgeLabel(() => ({}));
+
+  const nodeWidth = 172;
+  const nodeHeight = 36;
+
+  const getLayoutedElements = (elements, direction = 'TB') => {
```

Comment:
Right now we only use `TB`, so we could remove the `direction` parameter and all computations involving `LR`.